cpp/modmesh/view/RCameraController.cpp

```diff
@@ -0,0 +1,343 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <modmesh/view/RCameraController.hpp>
+
+#include <Qt3DInput/QKeyboardDevice>
+#include <Qt3DInput/QMouseDevice>
+#include <Qt3DRender/QCamera>
+
+namespace modmesh {
+
+RCameraInputListener::RCameraInputListener(
+    Qt3DInput::QKeyboardDevice* keyboardDevice, Qt3DInput::QMouseDevice* mouseDevice,
+    callback_type callback, QNode * parent)
+    : QEntity(parent)
+    , m_frameAction(new Qt3DLogic::QFrameAction)
+    , m_logicalDevice(new Qt3DInput::QLogicalDevice)
+
+    , m_keyboardDevice(keyboardDevice)
+    , m_mouseDevice(mouseDevice)
+    , m_callback(std::move(callback))
+
+    , m_rxAxis(new Qt3DInput::QAxis)
+    , m_ryAxis(new Qt3DInput::QAxis)
+    , m_txAxis(new Qt3DInput::QAxis)
+    , m_tyAxis(new Qt3DInput::QAxis)
+    , m_tzAxis(new Qt3DInput::QAxis)
+
+    , m_leftMouseButtonAction(new Qt3DInput::QAction)
+    , m_middleMouseButtonAction(new Qt3DInput::QAction)
+    , m_rightMouseButtonAction(new Qt3DInput::QAction)
+
+    , m_shiftButtonAction(new Qt3DInput::QAction)
+    , m_altButtonAction(new Qt3DInput::QAction)
+
+    , m_leftMouseButtonInput(new Qt3DInput::QActionInput)
+    , m_middleMouseButtonInput(new Qt3DInput::QActionInput)
+    , m_rightMouseButtonInput(new Qt3DInput::QActionInput)
+
+    , m_shiftButtonInput(new Qt3DInput::QActionInput)
+    , m_altButtonInput(new Qt3DInput::QActionInput)
+
+    , m_mouseRxInput(new Qt3DInput::QAnalogAxisInput)
+    , m_mouseRyInput(new Qt3DInput::QAnalogAxisInput)
+    , m_mouseTzXInput(new Qt3DInput::QAnalogAxisInput)
+    , m_mouseTzYInput(new Qt3DInput::QAnalogAxisInput)
+
+    , m_keyboardTxPosInput(new Qt3DInput::QButtonAxisInput)
+    , m_keyboardTyPosInput(new Qt3DInput::QButtonAxisInput)
+    , m_keyboardTzPosInput(new Qt3DInput::QButtonAxisInput)
+    , m_keyboardTxNegInput(new Qt3DInput::QButtonAxisInput)
+    , m_keyboardTyNegInput(new Qt3DInput::QButtonAxisInput)
+    , m_keyboardTzNegInput(new Qt3DInput::QButtonAxisInput)
+{
+    init();
+
+    connect(m_frameAction, &Qt3DLogic::QFrameAction::triggered,
+    this, [this] (const float dt) {
+        Qt3DExtras::QAbstractCameraController::InputState state{};
+
+        state.rxAxisValue = m_rxAxis->value();
+        state.ryAxisValue = m_ryAxis->value();
+        state.txAxisValue = m_txAxis->value();
+        state.tyAxisValue = m_tyAxis->value();
+        state.tzAxisValue = m_tzAxis->value();
+
+        state.leftMouseButtonActive = m_leftMouseButtonAction->isActive();
+        state.middleMouseButtonActive = m_middleMouseButtonAction->isActive();
+        state.rightMouseButtonActive = m_rightMouseButtonAction->isActive();
+
+        state.altKeyActive = m_altButtonAction->isActive();
+        state.shiftKeyActive = m_shiftButtonAction->isActive();
+
+        m_callback(state, dt);
```

Comment:
This is called every frame and it reads the input, and calls m_callback. m_callback is actually [updateCameraPositions](https://github.com/solvcon/modmesh/pull/385/files#diff-d22d8f512997b8a57a09a8179b7446c525a92a5797d54e5e8dfa0922555814bbR136) from camera controllers (orbit, and first person)